src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,277 @@
+            }
+
+            newVehicle.setVehicleID(stoi(entry.at(0)));
+            newVehicle.setHomeFireStation(findFireStation(stoi(entry.at(2))));
+            newVehicle.setMaxNumCrew(stoi(entry.at(3)));
+            newVehicle.setCurNumCrew(stoi(entry.at(4)));
+            newVehicle.setMaxWaterVolumeLitres(stof(entry.at(5)));
+            newVehicle.setCurWaterVolumeLitres(stof(entry.at(6)));
+            newVehicle.setWorkFactor(stof(entry.at(7)));
+
+            std::cout << "Adding vehicle #" << newVehicle.getVehicleID() << " to global list" <<std::endl;
+            districtVehicles.push_back(newVehicle);
+        }
+    }
+
+    void DistrictResources::processFireStationSampleData(const std::string &fileName)
+    {
+        std::vector<std::vector<std::string>> stations;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            stations.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : stations)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+
+            FireStation newStation{};
+
+            auto newLocation = new Location();
+            newLocation->setXCoord(stof(entry.at(1)));
+            newLocation->setYCoord(stof(entry.at(2)));
+
+            newStation.setFireStationID(stoi(entry.at(0)));
+            newStation.setFireStationLocation(newLocation);
+            newStation.setMaxNumCrew(stoi(entry.at(3)));
+            newStation.setCurNumCrew(stoi(entry.at(4)));
+
+            std::cout << "Adding Fire Station #" << newStation.getFireStationID() << " to global list" << std::endl;
+            districtFireStations.push_back(newStation);
+        }
+    }
+
+    void DistrictResources::processEventSampleData(const std::string &fileName, std::vector<Event> *pendingQueue)
+    {
+        std::vector<std::vector<std::string>> events;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            events.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : events)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+            std::string eventType = entry.at(3);
+            if (eventType == "CriticalSituation")
+            {
+                Event newEvent = *new Event();
```

Comment:
```suggestion
                Event newEvent();
```
Calling `new` heap-allocates. You dereference it on this line to obtain an object, but then the pointer is never saved anywhere, nor freed. The memory leak occurs on line 177, where the object is copied onto `pendingQueue` and the original `newEvent` is not freed.